NeoMathEngine/src/CPU/x86/avx/CpuMathEngineDnnConvAvxImpl.h

```diff
@@ -0,0 +1,697 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+// CBlobConvolution class specializations
+
+#if !defined( _mm256_set_m128 )
+// This instruction is defined since 8 gcc in avxintrin.h
+#define _mm256_set_m128( hi, lo) _mm256_insertf128_ps( _mm256_castps128_ps256( lo ), ( hi ), 0x1 )
+#endif
+
+namespace NeoML {
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Channel count: 24
+
+template<>
+inline CBlobConvolution<24>::CSize CBlobConvolution<24>::getWideBatchProcessSize()
+{
+	return { 1, 3 };
+}
+
+template<>
+inline void CBlobConvolution<24>::batchProcessChannels( const float* srcPtr, const float* fltPtr,
+	__m256& r00, __m256& r01, __m256& r02,
+	__m256& r10, __m256& r11, __m256& r12,
+	__m256& r20, __m256& r21, __m256& r22 )
+{
+	// Process three result pixels per time.
+	auto ProcessNext = [&]()
```

Comment:
Why lambda?  Because you can?

Reply:
Because I cannot completely abandon lambda expressions. Here the lambda is consistent with the implementation of other similar functions.